react-quill/app/routes/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from "react";
+import { Form } from "@remix-run/react";
+import { ActionFunction } from "@remix-run/node";
+import { ClientOnly } from "~/components/client-only";
+import { TextEditor } from "~/components/textEditor.client";
+
+import stylesheetQuill from "react-quill/dist/quill.snow.css";
+
+export function links() {
+  return [{ rel: "stylesheet", href: stylesheetQuill }];
+}
+
+export const action: ActionFunction = async ({ request }) => {
+  const form = await request.formData();
+
+  const textEditorValue = form.get("textEditor");
+  console.log(textEditorValue, "✅ Editor value is available to our server");
+  return "";
+};
+
+export default function Index() {
+  const [textEditor, setTextEditor] = useState<string>("");
```

Comment:
String type can be inferred

```suggestion
  const [textEditor, setTextEditor] = useState("");
```